### DISCUSSION TITLE
v3.1.6

### DISCUSSION THREAD

**Discussion by assafelovic** [assafelovic/gpt-researcher#1035] [Announcements]
Another release of performance improvements and bug fixes. Welcome @kga245 as a new contributor with a super valuable contribution and special shoutout again to @winsonluk for helping improve GPTR's stability!

## What's Changed
* Fix xAI bug by @winsonluk in https://github.com/assafelovic/gpt-researcher/pull/1021
* added language support by @assafelovic in https://github.com/assafelovic/gpt-researcher/pull/1026
* fix: Add language parameter support to generate_subtopic_report_prompt by @onlydole in https://github.com/assafelovic/gpt-researcher/pull/1029
* Fix generate_custom_report_prompt language bug by @winsonluk in https://github.com/assafelovic/gpt-researcher/pull/1033
* Introduce strategic_token_limit to fix Anthropic bug by @winsonluk in https://github.com/assafelovic/gpt-researcher/pull/1032
* Feature: unified logs handler by @kga245 in https://github.com/assafelovic/gpt-researcher/pull/1031

## New Contributors
* @kga245 made their first contribution in https://github.com/assafelovic/gpt-researcher/pull/1031

**Full Changelog**: https://github.com/assafelovic/gpt-researcher/compare/v3.1.5...v3.1.6

<hr /><em>This discussion was created from the release <a href='https://github.com/assafelovic/gpt-researcher/releases/tag/v3.1.6'>v3.1.6</a>.</em>

**Comment by kga245** [comment]
Glad to be a part of this. Thanks for the mention @assafelovic
